### PR TITLE
Git docs

### PR DESCRIPTION
**Pull request by yandthj**
Issue #301 

Ported and reviewed CMS git content and integrated with existing github pages git content.

### CODE REVIEW

File: docs/Documentation/Development/Build_Tools/git.md [hunk 1]
```diff
@@ -4,20 +4,196 @@ title: Git
...
+    git push origin main
+    ```
+
+??? note "Revert to a previous commit"
+    Find the commit you want to reset to:  
+    `git log`  
+    Once you have the hash:  
+    `git reset --hard <hash>`  
+    And to push onto the remote:  
+    `git push -f <remote> <branch>`
```
Comment (mcfarljm) [inline docs/Documentation/Development/Build_Tools/git.md:109]: Depending on the intent of this blurb, I would suggest either removing this example or using a different approach.  I would be hesitant to include a `git reset --hard` in a general guidance doc, as that can cause one to lose data (history), and should be used very cautiously.  And I definitely would not include an example with force pushing, as that should almost never be done unless someone really knows what they are doing (it overwrites history on a remote).

If the idea here is to undo the changes from a particular commit, there is `git revert`, which doesn't change history and is safer than `reset --hard`.  Or another approach is to create a new branch as a way of going back to a previous point in history, e.g., `git branch <branchname> <commit>`.

But once the changes in question get into a public version of the repo, history should not be changed with a force push.  So undoing public changes would mean either using git revert or manually creating new commits to undo the changes.

**Comment by mcfarljm** [comment]
Were the virtual GL and Fast X updates intended to be included as part of this PR?

File: docs/Documentation/Development/Build_Tools/git.md [hunk 2]
```diff
@@ -4,20 +4,196 @@ title: Git
...
+*Learn how to set up and use the Git software tool for development on the HPC systems*
+
+Git is used locally to track incremental development and modifications to a collection of files. [GitHub](https://github.com) is a git-repository hosting web-service, which serves as a synchronized, common access point for the file collections. GitHub also has social aspects, like tracking who changed what and why. There are other git hosting services like [GitLab](https://gitlab.com) which are similar to GitHub but offer slightly different features.
+
+
+NREL has a Github Enterprise server (github.nrel.gov) for internally-managed repos. Please note that github.nrel.gov is only available interally using the NREL network or VPN. NREL's git server uses SAML/SSO for logging into GitHub Enterprise. To get help accessing the server or creating a repository, please contact NREL ITS.
```
Comment (mcfarljm) [inline docs/Documentation/Development/Build_Tools/git.md:14]: Spelling:
```suggestion
NREL has a Github Enterprise server (github.nrel.gov) for internally-managed repos. Please note that github.nrel.gov is only available internally using the NREL network or VPN. NREL's git server uses SAML/SSO for logging into GitHub Enterprise. To get help accessing the server or creating a repository, please contact NREL ITS.
```

File: docs/Documentation/Development/Build_Tools/git.md [hunk 3]
```diff
@@ -4,20 +4,196 @@ title: Git
...
+
+???+ note "Set up SSH Key"
+    1. On the HPC system, copy the content of ~/.ssh/id_rsa.pub. 
+    1. On Github, click on: your git profile >  Settings > SSH and GPG keys > New SSH key
+    1. Paste the content of ~/.ssh/id_rsa.pub into the "Key" window
+    1. In your local git repo directory, type:
+    ```
+    git remote set-url origin <git@github.nrel.gov:username/my-projectname.git>.
+    ```
+    Your URL can be retrived in the Github UI by going to the remote repo, then "Code" > "SSH".
```
Comment (mcfarljm) [inline docs/Documentation/Development/Build_Tools/git.md:40]:
```suggestion
    Your URL can be retrieved in the Github UI by going to the remote repo, then "Code" > "SSH".
```

File: docs/Documentation/Development/Build_Tools/git.md (hunk 1 above)
Comment (yandthj) [reply docs/Documentation/Development/Build_Tools/git.md:109]: That's a good point. I wasn't the original author of this section, so I'm not sure of the intent, but changed to include `git revert` instead. I agree that's a better method.

File: docs/Documentation/Development/Build_Tools/git.md [hunk 4]
```diff
@@ -101,12 +101,12 @@ Users already have SSH keys created on the HPC systems. To set up Github SSH aut
     ```
 
 ??? note "Revert to a previous commit"
```
Comment (mcfarljm) [inline docs/Documentation/Development/Build_Tools/git.md:103]: The title of this note should probably be updated as well.  Could say either "Undo a previous commit" or "Revert a previous commit" (as opposed to reverting *to* a previous commit).

**Review by mcfarljm — APPROVED** [review]
LGTM.  Totally up to you, but given that all of the example blocks are very short, and that they make up basically all of the content of the page, I would find it more accessible/readable if they were not collapsed (e.g., `!!!` style or even default expanded with `???+`).

**Pull request merged**